co-log-core/src/Colog/Core/Class.hs

```diff
@@ -18,11 +17,9 @@ TODO: laws
 class HasLog env msg m where
     getLogAction :: env -> LogAction m msg
     setLogAction :: LogAction m msg -> env -> env
+    overLogAction :: (LogAction m msg -> LogAction m msg) -> env -> env
```

Comment:
Yes, you should add default implementations of `overLogAction` and `setLogAction`

Reply:
Shouldn't we just define one in terms of the other?
Is there a sensible default implementation for *both* at once?

Reply:
By default I mean no `default` keyword, just like for `Eq` typeclass where `==` is implemented using `/=` and `/=` is implemented using `==`